openpype/hosts/fusion/api/plugin.py

```diff
@@ -0,0 +1,262 @@
+
+        self._add_instance_to_context(instance)
+
+        return instance
+
+    def collect_instances(self):
+        comp = get_current_comp()
+        tools = comp.GetToolList(False, "Saver").values()
+        for tool in tools:
+            data = self.get_managed_tool_data(tool)
+            if not data:
+                continue
+
+            # Add instance
+            created_instance = CreatedInstance.from_existing(data, self)
+
+            # Collect transient data
+            created_instance.transient_data["tool"] = tool
+
+            self._add_instance_to_context(created_instance)
+
+    def update_instances(self, update_list):
+        for created_inst, _changes in update_list:
+            new_data = created_inst.data_to_store()
+            tool = created_inst.transient_data["tool"]
+            self._update_tool_with_data(tool, new_data)
+            self._imprint(tool, new_data)
+
+    def remove_instances(self, instances):
+        for instance in instances:
+            # Remove the tool from the scene
+
+            tool = instance.transient_data["tool"]
+            if tool:
+                tool.Delete()
+
+            # Remove the collected CreatedInstance to remove from UI directly
+            self._remove_instance_from_context(instance)
+
+    def _imprint(self, tool, data):
+        # Save all data in a "openpype.{key}" = value data
+
+        # Instance id is the tool's name so we don't need to imprint as data
+        data.pop("instance_id", None)
+
+        active = data.pop("active", None)
+        if active is not None:
+            # Use active value to set the passthrough state
+            tool.SetAttrs({"TOOLB_PassThrough": not active})
+
+        for key, value in data.items():
+            tool.SetData(f"openpype.{key}", value)
+
+    def _update_tool_with_data(self, tool, data):
+        """Update tool node name and output path based on subset data"""
+        if "subset" not in data:
+            return
+
+        original_subset = tool.GetData("openpype.subset")
+        original_format = tool.GetData(
+            "openpype.creator_attributes.image_format"
+        )
+
+        subset = data["subset"]
+        if (
+            original_subset != subset
+            or original_format != data["creator_attributes"]["image_format"]
+        ):
+            self._configure_saver_tool(data, tool, subset)
+
+    def _configure_saver_tool(self, data, tool, subset):
+        formatting_data = deepcopy(data)
+
+        # get frame padding from anatomy templates
+        anatomy = Anatomy()
+        frame_padding = anatomy.templates["frame_padding"]
+
+        # get output format
+        ext = data["creator_attributes"]["image_format"]
+
+        # Subset change detected
+        workdir = os.path.normpath(legacy_io.Session["AVALON_WORKDIR"])
+        formatting_data.update(
+            {"workdir": workdir, "frame": "0" * frame_padding, "ext": ext}
+        )
```

Comment:
```suggestion
        formatting_data.update({
            "workdir": workdir,
            "frame": "0" * frame_padding,
            "ext": ext,
            "product": {
                "name": formatting_data["subset"],
                "type": formatting_data["family"],
            },
        })
```

Reply:
This fix probably resolves https://github.com/ynput/OpenPype/pull/6057#pullrequestreview-1802621019